moveit_ros/moveit_servo/src/servo_node.cpp

```diff
@@ -214,12 +214,12 @@ std::optional<KinematicState> ServoNode::processJointJogCommand()
   if (!command_stale)
   {
     JointJogCommand command{ latest_joint_jog_.joint_names, latest_joint_jog_.velocities };
-    next_joint_state = servo_->getNextJointState(command);
+    next_joint_state = servo_->getNextJointState(robot_state, command);
   }
   else
   {
     auto result = servo_->smoothHalt(last_commanded_state_);
-    new_joint_jog_msg_ = result.first;
+    new_joint_jog_msg_ = !result.first;
```

Comment:
Was this a previous bug? Could you explain why this has a `!` in front of it now?

Reply:
Yes, I believe that was a bug. `smoothHalt` returns true if the joints are stopped. Until the joints are actually stopped, `new_joint_jog_msg_` should remain true so that `smoothHalt` keeps getting called. At some point, the joints will be stopped, which means `!result.first` will be false, `new_joint_jog_msg_` will be false, and the `processJointJogCommand` will stop getting called.